src/DSharp.Script/src/System/Properties.js

```diff
@@ -26,4 +26,15 @@ function defineProperty(instance, propertyName) {
         get: function () { return prop; },
         set: function (value) { prop = value; }
     });
+}
+function initializeObject(obj, initializerMap) {
+    if (!isValue(obj) || !isValue(initializerMap)) {
+        return obj;
+    }
+
+    for (var prop in initializerMap) {
```

Comment:
I would use object assign here, but it's an es6 api which isn't supported on some key browser targets :(